core/controlplane/config/config.go

```diff
@@ -435,6 +438,7 @@ type DeploymentSettings struct {
 	CloudWatchLogging       `yaml:"cloudWatchLogging,omitempty"`
 	AmazonSsmAgent          `yaml:"amazonSsmAgent,omitempty"`
 	CloudFormationStreaming bool `yaml:"cloudFormationStreaming,omitempty"`
+	KubeDnsMasq             `yaml:"kubeDnsMasq,omitempty"`
```

Comment:
itsn't it too much of a detail for no so much important feature? all dnsmasq should be configured equally, so just a flag under kube-dns should be enough, no? Otherwise  depth of config is comparable with nodepools :)